app/models/flood_risk_engine/comment.rb

```diff
@@ -0,0 +1,6 @@
+module FloodRiskEngine
+  class Comment < ActiveRecord::Base
+    belongs_to :commentable, polymorphic: true
+    belongs_to(:user) if defined?(User)
```

Comment:
The `if` is there, because the User model is defined in the BO only. So in other environments, the `belongs_to` needs to be disabled.

Also the `user` method is not being used to control access, but to record who added the comment.